include/mbedtls/x509_crt.h

```diff
@@ -348,6 +373,24 @@ int mbedtls_x509_crt_parse_file( mbedtls_x509_crt *chain, const char *path );
 int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path );
 #endif /* MBEDTLS_FS_IO */
 
+/**
+ * \brief          Extracts the othername information from a given
+ *                 subject alternative name.
+ *
+ * \param subject_alt_name      The subject alternative name to extract
+ *                              the othername information.
+ * \param other_name            The extracted othername information,
+ *                              as defined in RFC 5280 section 4.2.1.6.
+ *
+ * \note           Currently only "othername" of type hardware_module_name,
+ *                 as defined in RFC 4180 is supported.
+ *
+ * \return         MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE if the other name
+ *                 is of unsupported type_id. Zero for success and negative
+ *                 value for any other failure.
+ */
+int mbedtls_x509_get_other_name( const mbedtls_x509_sequence *subject_alt_name,
```

Comment:
Isnt it better to have the first parameter to be of type `const mbedtls_x509_crt`. Otherwise the user will have to look at the definition of mbedtls_x509_crt to get the subject_alt_name which we probably do not want to encourage

Reply:
ok

Reply:
This function is also called from the static function `x509_info_subject_alt_name()`, which doesn't hold the certificate. I'll find a way to walk around this.

Reply:
Actually, as I docuemtned, this function doesn't receive the whole subject alternative name extension, but a specific item from the SAN extension, which has `otherName` tag. It parses the otherName to the supported types, so it's not reasonable to have the certificate as a parameter.
Unless we redesign this function

Reply:
Well, I dont think its a good practice to force the programmer to look inside the cert struct to fish out the subject_alt_name. In fact, if this language had more object oriented support, then that field would probably be labelled private.

> Actually, as I docuemtned, this function doesn't receive the whole subject alternative name extension, but a specific item from the SAN extension, which has otherName tag. It parses the otherName to the supported types, so it's not reasonable to have the certificate as a parameter.

I do not see any problems with modifying the function slightly to unpack the structure internally and use whatever it needs to from the cert struct. The current design basically places this burden on the user.